### PR TITLE
Calculate flow through switches

### PR DESCRIPTION
**Pull request by marqueslanauja**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**
Utility to calculate flow through switches by` voltageLevel
`

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Review by MioRtia — COMMENTED** [review]
Thank you for your work! Seems good to me, only have one remark. I need to do some testing on how long it takes for a real test case too.

**Comment by marqueslanauja** [comment]
> Thank you for your work! Seems good to me, only have one remark. I need to do some testing on how long it takes for a real test case too.

In the last meeting Anne asked about the performance. I will prepare a table that shows us the performance for each IOP of 2021 March.

**Comment by zamarrenolm** [comment]
When exporting CGMES State Variables, including flows for all switches in the Network may have a big impact in the performance. The main impact is due to having to write lots of `SvPowerFlow` objects in the XML output, not for the calculation of flows in every voltage level.

**Review by zamarrenolm — APPROVED** [review]

File: iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/SwitchesFlow.java [hunk 1]
```diff
@@ -0,0 +1,429 @@
...
+            this.pflow = this.pflow + pFlow;
+            this.qflow = this.qflow + qFlow;
+
+            return this;
+        }
+    }
+
+    private static final class SwEdge extends DefaultWeightedEdge {
+        private final transient Switch sw;
+        private final transient VoltageLevel.NodeBreakerView.InternalConnection ic;
```
[the remark this answers is not in the document]
Comment (marqueslanauja) [reply iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/SwitchesFlow.java:372]: The edge is used by the graph component and probably each edge must be different. As there are edges associated with switches and internalConnections I prefer to keep the internalConnection, at least, for debugging purposes.

**Review by MioRtia — APPROVED** [review]

**Comment by annetill** [comment]
Do not forget to add documentation in the web-site for next release.

**Pull request merged**